### PR TITLE
1 regression tests available device #3335

### PR DESCRIPTION
**Pull request by BanzaiTokyo**
test_canberra_metric.py 
test_fractional_absolute_error.py 
test_fractional_bias.py 
test_geometric_mean_absolute_error.py

### CODE REVIEW

File: ignite/metrics/regression/pearson_correlation.py [hunk 1]
```diff
@@ -119,5 +119,12 @@ def compute(self) -> float:
...
         y_var = torch.clamp(y_var, min=0.0)
 
-        r = cov / torch.clamp(torch.sqrt(y_pred_var * y_var), min=self.eps)
+        denom = y_pred_var * y_var
+        denom = torch.clamp(denom, min=self.eps)
+        denom = torch.sqrt(denom)
+        r = cov / denom
+
+        if torch.isnan(r):
+            return 0.0
```
Comment (vfdev-5) [inline ignite/metrics/regression/pearson_correlation.py:128]: Revert this, I do not think this should be done right now. I also do not see a test updates for this metric

File: tests/ignite/metrics/regression/test_canberra_metric.py [hunk 2]
```diff
@@ -20,83 +20,83 @@ def test_wrong_input_shapes():
...
 
-    m = CanberraMetric()
+    m = CanberraMetric(device=available_device)
+    assert m._device == torch.device(available_device)
 
     canberra = DistanceMetric.get_metric("canberra")
 
-    m.update((torch.from_numpy(a), torch.from_numpy(ground_truth)))
-    np_sum = (np.abs(ground_truth - a) / (np.abs(a) + np.abs(ground_truth))).sum()
+    m.update((a, ground_truth))
```
Comment (vfdev-5) [inline tests/ignite/metrics/regression/test_canberra_metric.py:35]: let's do a for-loop as you do in tests/ignite/metrics/regression/test_fractional_absolute_error.py

File: tests/ignite/metrics/regression/test_fractional_absolute_error.py [hunk 3]
```diff
@@ -28,77 +28,65 @@ def test_wrong_input_shapes():
...
+    data = list(range(y_pred.shape[0] // batch_size))
+    fab = engine.run(data, max_epochs=1).metrics["fab"]
 
-    def get_test_cases():
-        test_cases = [
-            (torch.rand(size=(100,)), torch.rand(size=(100,)), 10),
-            (torch.rand(size=(100, 1)), torch.rand(size=(100, 1)), 20),
-        ]
-        return test_cases
+    np_sum = (2 * np.abs(np_y_pred - np_y) / (np.abs(np_y_pred) + np.abs(np_y))).sum()
```
Comment (vfdev-5) [inline tests/ignite/metrics/regression/test_fractional_absolute_error.py:86]: Let's use torch everywhere such that we could also remove `import numpy as np`

File: tests/ignite/metrics/regression/test_geometric_mean_absolute_error.py [hunk 4]
```diff
@@ -28,47 +28,48 @@ def test_wrong_input_shapes():
...
 
-    m = GeometricMeanAbsoluteError()
-    m.update((torch.from_numpy(a), torch.from_numpy(ground_truth)))
+    m = GeometricMeanAbsoluteError(device=available_device)
+    assert m._device == torch.device(available_device)
+    m.update((a, ground_truth))
 
-    errors = np.abs(ground_truth - a)
-    np_prod = np.multiply.reduce(errors) * np_prod
+    errors = torch.abs(ground_truth - a)
```
Comment (vfdev-5) [inline tests/ignite/metrics/regression/test_geometric_mean_absolute_error.py:43]: Let's write this using a for-loop

File: tests/ignite/metrics/regression/test_fractional_absolute_error.py (hunk 3 above)
Comment (BanzaiTokyo) [reply tests/ignite/metrics/regression/test_fractional_absolute_error.py:86]: done, but numpy is still used in _test_distrib_compute

Comment (vfdev-5) [reply tests/ignite/metrics/regression/test_fractional_absolute_error.py]: You can do the same there: `error = 2 * torch.abs(pred - ground_truth) / (torch.abs(pred) + torch.abs(ground_truth))`

Comment (BanzaiTokyo) [reply tests/ignite/metrics/regression/test_fractional_absolute_error.py]: sure, but wouldn't it make more sense to do a separate PR to do a pass on distributed tests?
